tests/models/test_openai.py

```diff
@@ -1915,6 +1915,111 @@ class MyModel(BaseModel):
     )
 
 
+def test_openai_transformer_with_recursive_ref() -> None:
+    """Test that OpenAIJsonSchemaTransformer correctly handles recursive models with $ref root."""
+    from pydantic_ai.profiles.openai import OpenAIJsonSchemaTransformer
+
+    # Create a schema with $ref root (recursive model scenario)
+    schema: dict[str, Any] = {
+        '$ref': '#/$defs/MyModel',
+        '$defs': {
+            'MyModel': {
+                'type': 'object',
+                'properties': {'foo': {'type': 'string'}},
+                'required': ['foo'],
+            },
+        },
+    }
+
+    transformer = OpenAIJsonSchemaTransformer(schema, strict=True)
+    result = transformer.walk()
+
+    # The transformer should resolve the $ref and use the transformed schema from $defs
+    # (not the original self.defs, which was the bug we fixed)
+    assert isinstance(result, dict)
+    # In strict mode, all properties should be required
+    assert 'properties' in result
+    assert 'required' in result
+    # The transformed schema should have strict mode applied (additionalProperties: False)
+    assert result.get('additionalProperties') is False
+    # All properties should be in required list (strict mode requirement)
+    assert 'foo' in result['required']
+
+
+def test_openai_transformer_fallback_when_defs_missing() -> None:
+    """Test fallback path when root_key is not in result['$defs'] (line 165).
+
+    This tests the safety net fallback that shouldn't happen in normal flow.
+    The fallback uses self.defs (original schema) when the transformed $defs
+    doesn't contain the root_key. This edge case is simulated using a mock.
+    """
+    from unittest.mock import patch
+
+    from pydantic_ai.profiles.openai import OpenAIJsonSchemaTransformer
+
+    schema: dict[str, Any] = {
+        '$ref': '#/$defs/MyModel',
+        '$defs': {
+            'MyModel': {
+                'type': 'object',
+                'properties': {'foo': {'type': 'string'}},
+                'required': ['foo'],
+            },
+        },
+    }
+
+    transformer = OpenAIJsonSchemaTransformer(schema, strict=True)
+
+    # Simulate edge case: super().walk() returns $defs without root_key
+    # This shouldn't happen in normal flow, but we test the fallback path
+    with patch.object(
+        transformer.__class__.__bases__[0],
```

Comment:
I don't feel comfortable with this; if we can't come up with a real schema that gets us to that line, do we need it at all?